pkg/generate/code/set_resource_test.go

```diff
@@ -3273,6 +3273,336 @@ func TestSetResource_EC2_DHCPOptions_NestedSetConfig(t *testing.T) {
 	)
 }
 
+func TestSetResource_EventBridge_Rule_SetResourceForStruct(t *testing.T) {
+	assert := assert.New(t)
+	require := require.New(t)
+
+	g := testutil.NewModelForService(t, "eventbridge")
+
+	crd := testutil.GetCRDByName(t, g, "Rule")
+	require.NotNil(crd)
+
+	// Getting KR api Target shape
+	field, ok := crd.SpecFields["Targets"]
+	require.True(ok)
+	krTargetShape := &field.ShapeRef.Shape.MemberRef
+
+	// Getting SDK Target shape
+	putTargetsOP, ok := g.SDKAPI.API.Operations["PutTargets"]
+	require.True(ok)
+	sdkTargetsShape, ok := putTargetsOP.InputRef.Shape.MemberRefs["Targets"]
+	require.True(ok)
+	sdkTargetShape := &sdkTargetsShape.Shape.MemberRef
+
+	expected := `	if sdkTarget.Arn != nil {
+		krTarget.ARN = sdkTarget.Arn
+	}
+	if sdkTarget.BatchParameters != nil {
+		krTargetf1 := &svcapitypes.BatchParameters{}
```

Comment:
Normally, this would just be:

```go
f1 := &svcapitypes.BatchParameters{}
```

not

```go
krTargetf1 := &svcapitypes.BatchParameters{}
```

I think you are incorrectly specifying the target variable name...

Reply:
The thing is if i change `krTarget` to an empty string, L3298 will become `.ARN = resp.Arn` which is not correct